integration-tests/springmvc-tests/src/test/java/org/apache/servicecomb/demo/springmvc/tests/SpringMvcIntegrationTestBase.java

```diff
@@ -477,6 +659,13 @@ public void ableToSetCustomHeader() {
         String.class);
 
     assertThat(jsonOf(result.getBody(), String.class), is("hei world"));
+
+    ListenableFuture<ResponseEntity<String>> listenableFuture = asyncRestTemplate.exchange(controllerUrl + "sayhei",
+        GET,
+        requestEntity,
+        String.class);
+    ResponseEntity<String> responseEntity = listenableFuture.get();
```

Comment:
all testcase is future.get, this is not the full async mode, please make sure async callback can work.

Reply:
ListenableFuture only support isDone, isCancelled and addCallback manually. add this case to test future.addCallbak in integration-test, ut and samples.